cg/apps/demultiplex/sample_sheet/api.py

```diff
@@ -0,0 +1,166 @@
+import logging
+import os
+from pathlib import Path
+
+import click
+from pydantic import ValidationError
+
+from cg.apps.demultiplex.sample_sheet.create import create_sample_sheet_content
+from cg.apps.demultiplex.sample_sheet.sample_models import FlowCellSample
+from cg.apps.demultiplex.sample_sheet.sample_sheet_validator import SampleSheetValidator
+from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.apps.lims import LimsAPI
+from cg.apps.lims.sample_sheet import get_flow_cell_samples
+from cg.constants.constants import FileFormat
+from cg.constants.demultiplexing import BclConverter
+from cg.exc import FlowCellError, HousekeeperFileMissingError, SampleSheetError
+from cg.io.controller import WriteFile, WriteStream
+from cg.meta.demultiplex.housekeeper_storage_functions import (
+    add_and_include_sample_sheet_path_to_housekeeper,
+)
+from cg.models.flow_cell.flow_cell import FlowCellDirectoryData
+
+LOG = logging.getLogger(__name__)
+
+
+class SampleSheetAPI:
+    """Sample Sheet API class."""
+
+    def __init__(self, flow_cell_dir: str, hk_api: HousekeeperAPI, lims_api: LimsAPI) -> None:
+        self.flow_cell_runs_dir = Path(flow_cell_dir)
+        self.hk_api: HousekeeperAPI = hk_api
+        self.lims_api: LimsAPI = lims_api
+        self.dry_run: bool = False
+        self.force: bool = False
+        self.validator = SampleSheetValidator()
+
+    def set_dry_run(self, dry_run: bool) -> None:
```

Comment:
Remove these setters and set the properties directly where they are used.
Getters and setters are not pythonic

Reply:
I have seen setters many times in the repository. I will leave them as they are for now and propose a discussion about them here https://github.com/Clinical-Genomics/atlas/pull/2377